torch/csrc/jit/autodiff.cpp

```diff
@@ -119,9 +119,13 @@ static std::vector<Value*> gradientForNode(Node* node, ArrayRef<Value*> grad_val
 
     } else if (node->matches("aten::clamp(Tensor self, Scalar min, Scalar max) -> Tensor")) {
       // we do two type_as as it's free (hopefully) and the "*" only works with float
-      return {grads.at(0)
-	      * (inputs.at(0) > inputs.at(1)).type_as(inputs.at(0))
-	      * (inputs.at(0) < inputs.at(2)).type_as(inputs.at(0)), nullptr, nullptr};
+      // the "! (val > min)" is chosen such that the gradient is 0 on the
+      // boundary and the factor is 1 when the boundary is NaN
+      // the ! is expressed as "1-" for lack of a "not" function and
+      // the the fuser insisting on float
+      return {(inputs.at(0).isnan() ? inputs.at(0) : grads.at(0))
```

Comment:
I don't think it will work, it's not an elementwise traversal. isnan() returns tensor which will always evaluate to true, so you'll always be getting inputs.at(0) here.

Reply:
@ngimel Thanks for pointing that out! The sweet poison of overloaded symbolic operators made me think that anything would work...
@vishwakftw I'm trying to stick with fusable operators. I think at::where might count eventually, but only when you can have byte inputs which I understand you cannot at the moment.

Reply:
You can have byte inputs, but you'd have to add some special casing in the autodiff pass and fusion pass.

Reply:
Fusing where might be attractive, then...